File: meet-in-docomo/Smart-in/Client/Callback/Sample/php/smartin.php
<?php
/*	Smart-inへの接続要求、およびステータスチェックを行います。
*	以下に必要情報を記載してからご利用ください。
*	{Smart-inキー32桁}
*	{Smart-in企業コード4桁}
*	{DB名}
*	{DBユーザ名}
*	{DBパスワード}
*	{設置DIR}
*/

// sin_crypt 暗号化キー（半角英数32桁）
define('KEY_CODE', '{Smart-inキー32桁}');
// 企業コード
define('COMPANY_CODE', '{Smart-in企業コード4桁}');
// 依頼区分(要求された電話番号に発信を行い、コールバック認証を行う。)
define('REQUEST_CODE', 'C51');
// Smart-in 依頼用サーバ
define('SMARTIN_SVR', 'api.smart-in.biz');	// こちらは試験用です。提供環境に応じ、適宜変更ください。
// sin_crypt 格納パス
define('SIN_CRYPT_PATH', './sin_crypt');
// 確認結果POST用URL
define('RESPONSE_URL', 'https://' . $_SERVER['HTTP_HOST'] . '{設置DIR}/smartin_response.php');	// URLはご利用の環境に応じ、ディレクトリを追記ください。

// DB設定
define('PDO_DSN', 'mysql:dbname={DB名}; host=localhost');
define('PDO_USER', '{DBユーザ名}');
define('PDO_PASS', '{DBパスワード}');

if ($_SERVER['REQUEST_METHOD'] == 'POST') {

	$mode = $_REQUEST['mode'];

	/*	
		Smart-inへコールバックリクエストを行う
	*/
	if ($mode == 'request') {

		// 接続先電話番号
		$tel_no = $_POST['tel_no'];
					
		// smart-inへのリクエストパラメータを生成 
		$request = array(
			'company' => COMPANY_CODE,
			'code' => REQUEST_CODE,
			'telno' => $tel_no,
			'response_url' => RESPONSE_URL
		);		 
		$requestJson = json_encode($request); 
					
		// jsonを専用APIで暗号化する 
		$execShellCommand = SIN_CRYPT_PATH." -e ".KEY_CODE." '${requestJson}'";
		exec($execShellCommand, $output, $encodeResult);

		// 暗号化成功
		if($encodeResult == 0){		 
			
			// Smart-inへの送信データに暗号化したjsonを代入
			$authJson = $output[0];
			$url = 'http://'.SMARTIN_SVR.'/request.cgi';
			$postdata = http_build_query(	   
				array(  
					'company' => COMPANY_CODE,
					'data' => $authJson
				)   
			);	  
			
			// http設定
			$opts = array('http' =>	 
				array(  
					'method'  => 'POST',
					'header'  => 'Content-type: application/x-www-form-urlencoded',
					'content' => $postdata
				)   
			);	  
			$context  = stream_context_create($opts);	   
					
			// Smart-inにリクエスト
			$response = file_get_contents($url, false, $context);	   
					
			// レスポンスをデコードし、画面に表示	 
			$execShellCommand = SIN_CRYPT_PATH." -d ".KEY_CODE." ${response}";
			$responseJson = exec($execShellCommand, $output, $decodeResult);  
			if($decodeResult == 0){
				$responseDecodeStr = "";
				for ($i = 1; $i < count($output)-1; $i++) {
					$responseDecodeStr .= $output[$i];
				}

				// デコードすると token と result が出てくる
				// result 0:認証中 1:認証完了 9:認証失敗
				$responseArray = json_decode($responseDecodeStr, true); 

				echo $responseArray['token'];

			// デコード失敗
			}else{	 
				echo 'デコード失敗';
			}

		// 暗号化失敗   
		}else{
			echo '暗号化失敗';
		}

		exit;
	}

	/*	
		Smart-inのコールバックから登録した情報を参照する
	*/
	if ($mode == 'check'){
		$token = $_POST['token'];
		if (strlen($token) != 32)
			exit;

		// 結果を取得（DB使用時）
		$dbh = new PDO(PDO_DSN, PDO_USER, PDO_PASS);
		$sql = "SELECT * FROM tokens WHERE token = '${token}'";
		$stmt = $dbh->query($sql);
		$array = $stmt->fetchAll(PDO::FETCH_ASSOC);
		if(count($array) > 0)
			$smartin_res = $array[0]['status'];
		$dbh = null;

		// 結果を取得（ファイル使用時）
		//$db = parse_ini_file('textdb.db');
		//$smartin_res = $db[$token];


		switch ($smartin_res) {
			case 100:
				$result = 'true';
				break;
			case 101: 
				$result = '本人話中';
				break;
			case 102: 
				$result = '着信時拒否';
				break;
			case 103: 
				$result = 'タイムアウト';
				break;
			case 109: 
				$result = '例外';
				break;
			default:
				$result = 'waiting';
				break;
		}

		echo $result;
		exit;
	}

}

File: meet-in-docomo/Smart-in/Client/Callback/Sample/C#/App_Code/smartin.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using System.Diagnostics;
using System.Web;
using System.Web.Script.Serialization;

/// <summary>
/// smartin の概要の説明です
/// </summary>
public class smartin
{

    // sin_crypt 暗号化キー（半角英数32桁）
    protected static string KEY_CODE = "{Smart-inキー32桁}";
    // 企業コード
    protected static string COMPANY_CODE = "{Smart-in企業コード4桁}";
    // 依頼区分(要求された電話番号に発信を行い、コールバック認証を行う。)
    protected static string REQUEST_CODE = "C51";
    // Smart-in 依頼用サーバ
    protected static string SMARTIN_SVR= "api.smart-in.biz";	// こちらは試験用です。提供環境に応じ、適宜変更ください。
    // sin_crypt 格納パス
    protected static string SIN_CRYPT_PATH = HttpContext.Current.Request.PhysicalApplicationPath + "sin_crypt.exe";
    // 確認結果POST用URL
    protected static string RESPONSE_URL = "https://" + HttpContext.Current.Request.Url.Host + "{設置DIR}" + "/smartin_response.aspx";	// URLはご利用の環境に応じ、ディレクトリを追記ください。

	public smartin()
	{
		//
		// TODO: コンストラクター ロジックをここに追加します
		//
	}

    /// <summary>
    /// 
    /// </summary>
    /// <param name="tel_no">接続先電話番号</param>
    /// <param name="token_str">トークン</param>
    /// <param name="message">メッセージ</param>
    /// <returns>処理結果</returns>
    public bool request(string tel_no, ref string token_str, ref string message)
    {
        // 処理結果
        bool result = false;

		// smart-inへのリクエストパラメータを生成 
        JavaScriptSerializer serializer = new JavaScriptSerializer();
        string json = serializer.Serialize(
            new {
                company = COMPANY_CODE,
                code = REQUEST_CODE,
                telno = tel_no,
                response_url = RESPONSE_URL
            }
            );
        json = "\"" + json.Replace("\"", "\"\"") + "\"";

        // jsonを専用APIで暗号化してえられた出力
        string batchOutput = "";

        // 暗号化結果を取得
        int batchResult = Utility.processShell(SIN_CRYPT_PATH, "-e " + KEY_CODE + " " + json, ref batchOutput);

		// 暗号化成功
		if(0 == batchResult) {		 
			// Smart-inへの送信データに暗号化したjsonを代入
			string authJson = batchOutput;
			string url = "http://" + SMARTIN_SVR + "/request.cgi";

            string postdata = "company=" + COMPANY_CODE + "&"
                            + "data=" + authJson;

            // Smart-inにリクエスト
            string response = Utility.postData(url, postdata);

            // jsonを専用APIで暗号化してえられた出力
            string responseJson = "";

			// レスポンスをデコードし、画面に表示	 
            int decodeResult = Utility.processShell(SIN_CRYPT_PATH, "-d " + KEY_CODE + " " + response, ref responseJson);
					
			if (0 == decodeResult) {
				// デコードすると token と result が出てくる
				// result 0:認証中 1:認証完了 9:認証失敗
                dynamic jsonObject = serializer.Deserialize<object>(responseJson);

				token_str = jsonObject["token"];

                result = true;
			} else 
			// デコード失敗
            {	 
				message = "デコード失敗";
			}
		// 暗号化失敗   
		} else {
			message = "暗号化失敗";
		}

        return result;
    }

    public bool check(string token, ref string message)
    {
		if (token.Length != 32) {
			return false;
        }

        long smartin_res = 0;

		// 結果を取得（DB使用時）
        SqlConnection connection = new SqlConnection();
        SqlCommand command = new SqlCommand();
        DataTable dt = new DataTable();

        connection.ConnectionString = ConfigurationManager.ConnectionStrings["connectionString"].ToString();
        connection.Open();

        command.CommandText = "SELECT * FROM tokens WHERE token = '" + token + "'";
        command.Connection = connection;

        SqlDataReader sqlDataReader = command.ExecuteReader();

        while (sqlDataReader.Read()) {
            smartin_res = (long)sqlDataReader["status"];
        }

        sqlDataReader.Close();
        command.Connection.Close();
        command.Dispose();
        connection.Close();
        connection.Dispose();

		switch (smartin_res) {
			case 100:
				message = "true";
				break;
			case 101: 
				message = "本人話中";
				break;
			case 102: 
				message = "着信時拒否";
				break;
			case 103: 
				message = "タイムアウト";
				break;
			case 109: 
				message = "例外";
				break;
			default:
				message = "waiting";
				break;
		}

        return true;
    }
}